python/revng/pypeline/schedule/schedule.py

```diff
@@ -244,3 +250,48 @@ def serialize(self) -> str:
             visited_tasks.append(task)
 
         return yaml.safe_dump({"containers": containers, "tasks": tasks})
+
+    def _identify_discardable_containers(self):
+        """
+        Given a schedule, set the ScheduleTasks with the containers that will
+        be discarded when the schedule is executed.
+        """
+
+        scheduled_task: ScheduledTask = self.root
+        # Assume that all the outgoing request of the root task are going to
+        # be read, so the caller of the `run` method is implicitly the last reader
+        readers_encountered: set[ContainerDeclaration] = {
+            cd for cd, objects in self.root.outgoing.items() if len(objects) != 0
+        }
+
+        # Inspect the tasks in backward order, from the last one to the first.
+        # The logic used is the following:
+        # * If a task reads a container last (either with READ or READ_WRITE)
+        #   then it should be marked as disposable
```

Comment:
* If a task is the last one to read (either with READ or READ_WRITE) a container, the container should be marked as disposable.
* If a task clobbers a container (with WRITE) then the container can be marked as disposable in the preceding task